src/core/test/system/borrower/switchable_distributor/withdraw_single_tranche.t.sol

```diff
@@ -0,0 +1,276 @@
+        juniorOperator.relyInvestor(juniorInvestor_);
+    }
+
+    function withdraw(uint loanId, uint tokenId, uint amount, address usr) public {
+        uint shelfBalance = currency.balanceOf(address(shelf));
+        uint juniorBalance = currency.balanceOf(address(junior));
+        uint initialAvailable = add(shelfBalance, juniorBalance);
+        uint initialRecipientBalance = currency.balanceOf(usr);
+        uint initialLoanBalance = shelf.balances(loanId);
+        uint initialTotalBalance = shelf.balance();
+        borrower.withdraw(loanId, amount, usr);
+        assertPostCondition(loanId, tokenId, amount, usr, initialAvailable, initialRecipientBalance, initialLoanBalance, initialTotalBalance);
+    }
+
+    function assertPreCondition(uint loanId, uint tokenId, uint amount) public {
+        // assert: borrower loanOwner
+        assertEq(title.ownerOf(loanId), borrower_);
+        // assert: shelf nftOwner
+        assertEq(collateralNFT.ownerOf(tokenId), address(shelf));
+        // assert: loan has enough balance 
+        assert(shelf.balances(loanId) >= amount);
+        // assert: enough funds available
+        uint shelfBalance = currency.balanceOf(address(shelf));
+        uint juniorBalance = currency.balanceOf(address(junior));
+        assert(add(shelfBalance, juniorBalance) >= amount);
+    }
+
+    function assertPostCondition(uint loanId, uint tokenId, uint withdrawAmount, address recipient, uint initialAvailable, uint initialRecipientBalance, uint initialLoanBalance, uint initialTotalBalance) public {
+        // assert: nft still locked, shelf nftOwner
+        assertEq(collateralNFT.ownerOf(tokenId), address(shelf));
+        // assert: available balance decreased
+        assertEq(currency.balanceOf(address(shelf)), sub(initialAvailable, withdrawAmount));
+        // assert: no money left in tranche reserve -> withdraw calls balance
+        assertEq(currency.balanceOf(address(junior)), 0);
+        // assert: borrower balance increased
+        assertEq(currency.balanceOf(recipient), add(initialRecipientBalance, withdrawAmount));
+        // assert: loan balance reduced
+        assertEq(shelf.balances(loanId), sub(initialLoanBalance, withdrawAmount));
+        // assert: total balance reduced
+        assertEq(shelf.balance(), sub(initialTotalBalance, withdrawAmount));
+    }
+
+    function testWithdraw() public {
+        uint loanAmount = 100 ether;
+        uint investAmount = loanAmount;
+        // issue nft for borrower
+        (uint tokenId, ) = issueNFT(borrower_);
+        // issue loan for borrower
+        uint loanId = borrower.issue(collateralNFT_, tokenId);
+        // lock nft
+        lockNFT(loanId, borrower_);
+        // init borrow -> add loan balance
+        initBorrow(loanId, tokenId, loanAmount, borrower_);
+        // junior investor puts money into tranche
+        invest(investAmount);
+        // move funds into shelf
+        distributor.balance(); // why not worky
```

Comment:
technically you don't need this call bc it should be called in withdraw already, i would take it out since you are testing withdraw